app/models/developer.rb

```diff
@@ -22,6 +22,11 @@ class Developer < ApplicationRecord
   validates :cover_image, content_type: ["image/png", "image/jpg", "image/jpeg", "image/gif"],
     max_file_size: 10.megabytes
 
+  validates_numericality_of :preferred_min_hourly_rate, less_than: ->(developer) { developer.preferred_max_hourly_rate }, if: -> { preferred_min_hourly_rate.present? }
+  validates_numericality_of :preferred_max_hourly_rate, greater_than: ->(developer) { developer.preferred_min_hourly_rate }, if: -> { preferred_max_hourly_rate.present? }
+  validates_numericality_of :preferred_min_salary, less_than: ->(developer) { developer.preferred_max_salary }, if: -> { preferred_min_salary.present? }
+  validates_numericality_of :preferred_max_salary, greater_than: ->(developer) { developer.preferred_min_salary }, if: -> { preferred_max_salary.present? }
```

Comment:
Let's change all of these to `_or_equal_to`, just in case someone _really_ has a specific rate/salary in mind.